crates/mun_language_server/tests/support.rs

```diff
@@ -1,96 +1,175 @@
-use async_std::future::timeout;
-use futures::{SinkExt, StreamExt};
-use lsp_types::{notification::Exit, request::Shutdown};
-use mun_language_server::protocol::{Connection, Message, Notification, Request};
-use mun_language_server::{main_loop, Config};
+use crossbeam_channel::{after, select};
+use lsp_server::{Connection, Message, Notification, Request};
+use lsp_types::{
+    notification::Exit, request::Shutdown, ProgressParams, ProgressParamsValue, WorkDoneProgress,
+};
+use mun_language_server::{main_loop, Config, FilesWatcher};
+use mun_test::Fixture;
 use paths::AbsPathBuf;
+use project::ProjectManifest;
 use serde::Serialize;
 use serde_json::Value;
-use std::convert::TryFrom;
+use std::cell::{Cell, RefCell};
+use std::convert::TryInto;
+use std::fs;
 use std::time::Duration;
 
+/// A `Project` represents a project that a language server can work with. Call the `server` method
```

Comment:
By adding `[..]` to the server, it should become a link to the actual server in docs